architecture-decisions/0004-preservation-fixity.md

```diff
@@ -0,0 +1,111 @@
+      }
+      ```
+      - Production Status Topic is configured with the following commands:
+        ```
+        gcloud beta pubsub topics create figgy-production-fixity-status
+        echo '{"bindings":[{"members":["serviceAccount:figgy-preservation-production@pulibrary-figgy-storage-1.iam.gserviceaccount.com"],"role":"roles/pubsub.editor"}],"etag":"ACAB"}' > permissions.json
+        gcloud beta pubsub topics set-iam-policy projects/pulibrary-figgy-storage-1/topics/figgy-production-fixity-status permissions.json
+        gcloud beta pubsub subscriptions create figgy-production-fixity-status --topic figgy-production-fixity-status --expiration-period=never
+        gcloud beta pubsub subscriptions set-iam-policy projects/pulibrary-figgy-storage-1/subscriptions/figgy-production-fixity-status permissions.json
+        rm permissions.json
+        ```
+      - Staging Status Topic is configured with the following commands:
+        ```
+        gcloud beta pubsub topics create figgy-staging-fixity-status
+        echo '{"bindings":[{"members":["serviceAccount:figgy-staging@pulibrary-figgy-storage-1.iam.gserviceaccount.com"],"role":"roles/pubsub.editor"}],"etag":"ACAB"}' > permissions.json
+        gcloud beta pubsub topics set-iam-policy projects/pulibrary-figgy-storage-1/topics/figgy-staging-fixity-status permissions.json
+        gcloud beta pubsub subscriptions create figgy-staging-fixity-status --topic figgy-staging-fixity-status --expiration-period=never
+        gcloud beta pubsub subscriptions set-iam-policy projects/pulibrary-figgy-storage-1/subscriptions/figgy-staging-fixity-status permissions.json
+        rm permissions.json
+        ```
+   1. A daemon will run on each Figgy worker machine which pulls these events
+      and sends them to Sidekiq. This worker can be executed via
+      `RAILS_ENV=production bundle exec rake fixity:run_worker`, but does not
+      manually need to be started - it is handled by a systemd service deployed
+      via Ansible.
+   1. Sidekiq will create `Event` objects for each status sent, which will be
+      displayed in the Fixity Dashboard.
+
+## Consequences
+
+1. Cost
+   1. Retrieval of resources from Google Coldline has a hefty cost associated
+      with it. If we find that 10% per year is too expensive for the benefits of
+      manual fixity checking we may want to scale that down.
+1. Random Checking
+   1. 10% a year leaves a lot of room for some resources to never be checked.
+      This is purely an investigative measure to ensure the reliability
+      guaruntees of Google Cloud Storage as well as validate our preservation process.
+      Should failures become evident, we will likely want to check and fix
+      everything.
+2. Complexity
+   1. The above system has a few moving parts which may be difficult to explain.
+      However, as we're dealing with many terabytes of data, it's the cheapest
+      and most time-efficient method of fixity checking that amount of material.
```

Comment:
I think it would be good to add something about Coldline vs. Nearline storage and retrieval costs, and the amount of fixity checking we do, being a somewhat complex calculation to figure out which is cheaper.